File: braggbay555_ui/src/services/ProductCategoryService.js
import http from "../http-common"; 

class ProductCategoryService {
  getAllProductCategorys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/productCategory/productCategorys`, searchDTO);
  }

  get(productCategoryId) {
    return this.getRequest(`/productCategory/${productCategoryId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/productCategory?field=${matchData}`, null);
  }

  addProductCategory(data) {
    return http.post("/productCategory/addProductCategory", data);
  }

  update(data) {
  	return http.post("/productCategory/updateProductCategory", data);
  }
  
  uploadImage(data,productCategoryId) {
  	return http.postForm("/productCategory/uploadImage/"+productCategoryId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ProductCategoryService();

File: src/main/java/com/braggbay555/service/ProductCategoryService.java
package com.braggbay555.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbay555.domain.ProductCategory;
import com.braggbay555.dto.ProductCategoryDTO;
import com.braggbay555.dto.ProductCategorySearchDTO;
import com.braggbay555.dto.ProductCategoryPageDTO;
import com.braggbay555.dto.ProductCategoryConvertCriteriaDTO;
import com.braggbay555.service.GenericService;
import com.braggbay555.dto.common.RequestDTO;
import com.braggbay555.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ProductCategoryService extends GenericService<ProductCategory, Integer> {

	List<ProductCategory> findAll();

	ResultDTO addProductCategory(ProductCategoryDTO productCategoryDTO, RequestDTO requestDTO);

	ResultDTO updateProductCategory(ProductCategoryDTO productCategoryDTO, RequestDTO requestDTO);

    Page<ProductCategory> getAllProductCategorys(Pageable pageable);

    Page<ProductCategory> getAllProductCategorys(Specification<ProductCategory> spec, Pageable pageable);

	ResponseEntity<ProductCategoryPageDTO> getProductCategorys(ProductCategorySearchDTO productCategorySearchDTO);
	
	List<ProductCategoryDTO> convertProductCategorysToProductCategoryDTOs(List<ProductCategory> productCategorys, ProductCategoryConvertCriteriaDTO convertCriteria);

	ProductCategoryDTO getProductCategoryDTOById(Integer productCategoryId);







}





